src/lib/homeobject_impl.hpp

```diff
@@ -38,15 +38,12 @@ inline shard_id_t make_new_shard_id(pg_id_t pg, shard_id_t next_shard) {
 
 struct Shard {
     explicit Shard(ShardInfo info) : info(std::move(info)) {}
+    virtual ~Shard() = default;
```

Comment:
Where is this used poly-morphically? It's not exposed above HomeObject, or below; so we shouldn't need a vtable.

Reply:
In hs_homeobject.hpp,  I had added a new derived class HS_Shard which will including some members such as HS superblock, which is only meaning to HS version homeobject and I do not want to expose these details to Shard level.

Reply:
I see; so it can be a `unique_ptr` then?

Reply:
yes, it is already updated.